samples/go/photo-index/main.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/attic-labs/noms/go/dataset"
+	"github.com/attic-labs/noms/go/spec"
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/go/walk"
+	flag "github.com/juju/gnuflag"
+)
+
+func main() {
+	if !index() {
+		os.Exit(1)
+	}
+}
+
+func index() (win bool) {
+	var dbStr = flag.String("db", "", "input database spec")
+	var outDSStr = flag.String("out-ds", "", "output dataset to write to - if empty, defaults to input dataset")
+
+	flag.Usage = usage
+	flag.Parse(false)
+
+	if flag.NArg() == 0 {
+		flag.Usage()
+		return
+	}
+
+	if flag.NArg() == 0 {
+		fmt.Fprintln(os.Stderr, "Need at least one dataset to index")
+		return
+	}
+
+	db, err := spec.GetDatabase(*dbStr)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid input database '%s': %s\n", flag.Arg(0), err)
+		return
+	}
+	defer db.Close()
+
+	var outDS dataset.Dataset
+	if !dataset.DatasetFullRe.MatchString(*outDSStr) {
+		fmt.Fprintf(os.Stderr, "Invalid output dataset name: %s\n", *outDSStr)
+		return
+	} else {
+		outDS = dataset.NewDataset(db, *outDSStr)
+	}
+
+	inputs := []types.Value{}
+	for i := 0; i < flag.NArg(); i++ {
+		if dataset.DatasetFullRe.MatchString(flag.Arg(i)) {
```

Comment:
Should we allow paths to be specified as well as datasets? This is useful, for example when I want do say <dsname>.value so all the history isn't scanned.